test/e2e/core/getSourceData.spec.js

```diff
@@ -0,0 +1,145 @@
+describe('Core.getSourceData', () => {
+  const id = 'testContainer';
+
+  beforeEach(function() {
+    this.$container = $(`<div id="${id}"></div>`).appendTo('body');
+  });
+
+  afterEach(function() {
+    if (this.$container) {
+      destroy();
+      this.$container.remove();
+    }
+  });
+
+  it('should return a copy of the dataset passed at init, instead of a reference', () => {
+    const datasetAoA = [
+      ['a1', 'a2', 'a3'],
+      ['b1', 'b2', 'b3'],
+      ['c1', 'c2', 'c3']
+    ];
+
+    const datasetAoO = [
+      { a: 1, b: 2, c: 3, d: { e: 'nested1', f: 'nested2_1' } },
+      { a: 4, b: 5, c: 6, d: { e: 'nested2', f: 'nested2_2' } },
+      { a: 7, b: 8, c: 9, d: { e: 'nested3', f: 'nested2_3' } }
+    ];
+
+    handsontable({
+      data: datasetAoA,
+      columns: [
+        {}
+      ]
+    });
+
+    getSourceData()[0][0] = 'changed!';
+
+    expect(getSourceData()[0][0]).toEqual('a1');
+
+    updateSettings({
+      data: datasetAoO
+    });
+
+    getSourceData()[0].a = 'changed!';
+
+    expect(getSourceData()[0].a).toEqual(1);
```

Comment:
Maybe an extra test with `toBe` will be helpful and will guarantee that it is not a reference?